modules/apps/transfer/keeper/grpc_query_test.go

```diff
@@ -140,3 +140,74 @@ func (suite *KeeperTestSuite) TestQueryParams() {
 	res, _ := suite.queryClient.Params(ctx, &types.QueryParamsRequest{})
 	suite.Require().Equal(&expParams, res.Params)
 }
+
+func (suite *KeeperTestSuite) TestQueryDenomHash() {
+
+	reqTrace := types.DenomTrace{
+		Path:      "transfer/channelToA/transfer/channelToB",
+		BaseDenom: "uatom",
+	}
+
+	var (
+		req     *types.QueryDenomHashRequest
+		expHash = reqTrace.Hash().String()
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"invalid trace",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: "transfer/channelToA/transfer/",
+				}
+			},
+			false,
+		},
+		{
+			"not found denom trace",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: reqTrace.GetFullDenomPath(),
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: reqTrace.GetFullDenomPath(),
+				}
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), reqTrace)
+
+				req = &types.QueryDenomHashRequest{
```

Comment:
This block of code is duplicated. Is it needed?

Reply:
@crodriguezvega No, my bad